orchagent/neighorch.cpp

```diff
@@ -541,6 +563,26 @@ void NeighOrch::doTask(Consumer &consumer)
 
         IpAddress ip_address(key.substr(found+1));
 
+        if (gPortsOrch->isInbandPort(alias))
+        {
+            Port ibport;
+            gPortsOrch->getInbandPort(ibport);
+            if(ibport.m_type == Port::VLAN)
+            {
+                /* For inband Vlan, a kernel neigh is added for every remote neighbor
+                 * and the interface that the neigh is attached is inband Vlan.
+                 * neighsyncd adds the kernel neigh to neigh table in APPL_DB again.
+                 * The following change is to skip adding these remote neighbor attached
+                 * to inband Vlan.
+                 */
+                if (m_remoteNeigh.find(ip_address) != m_remoteNeigh.end())
+                {
+                    it = consumer.m_toSync.erase(it);
+                    continue;
+                }
```

Comment:
This logic can go as else part (after line 560). The getPortOrch->isInbandPort(alias) and getPortsOrch->getInbandPort(ibport) are redundantly called multiple times.

Reply:
Thanks. Fixed.